### PR TITLE
NEWSWORLDSERVICE-2062: put zhongwen on CPS front page temporarily

### PR DESCRIPTION
**Pull request by eagerterrier**
temporary update to see zhongwen solution for optimo

### CODE REVIEW

File: src/app/lib/utilities/getStoryPromoInfo/index.jsx [hunk 1]
```diff
@@ -28,10 +28,15 @@ export const getHeadline = item => {
...
-export const getUrl = item => {
+export const getUrl = (item, service = null, variant = null) => {
   const assetUri = pathOr(null, ['locators', 'assetUri'], item);
   const canonicalUrl = pathOr(null, ['locators', 'canonicalUrl'], item);
-  const uri = pathOr(null, ['uri'], item);
+  let uri = pathOr(null, ['uri'], item);
+  if (uri && service === 'zhongwen') {
+    if (uri.indexOf(`/${variant}`) === -1 && uri.indexOf('/articles/') !== -1) {
+      uri = uri.substr(0, uri.length - variant.length) + variant;
+    }
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getStoryPromoInfo/index.jsx]: Could we use replace instead substring? (I suppose substring is arguably more secure?)

File: src/app/routes/utils/regex/index.test.js [hunk 2]
```diff
@@ -503,7 +503,9 @@ describe('frontPage -> homePage migration', () => {
 
   const serviceToRoute = service => `/${service}`;
 
-  const homepageServices = worldServices.map(serviceToRoute);
+  const homepageServices = worldServices
+    .filter(service => service !== 'zhongwen')
+    .map(serviceToRoute);
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/index.test.js]: We need to revert this before merging
```suggestion
  const homepageServices = worldServices.map(serviceToRoute);
```

File: src/app/routes/utils/regex/utils/index.js [hunk 3]
```diff
@@ -81,21 +81,25 @@ export const getFrontPageRegex = services => {
       service => !homePageServices.includes(service),
     );
   } else {
-    frontPages = [];
+    frontPages = ['zhongwen'];
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/utils/index.js]: Revert
```suggestion
    frontPages = [];
```

File: src/app/routes/utils/regex/utils/index.js [hunk 4]
```diff
@@ -81,21 +81,25 @@ export const getFrontPageRegex = services => {
...
   }
   const serviceRegex = getServiceRegex(frontPages);
   return `/:service(${serviceRegex}):variant(${variantRegex})?:amp(${ampRegex})?:lite(${liteRegex})?`;
 };
 
 export const getTipoHomeRegex = services => {
-  const serviceRegex = getServiceRegex(services);
+  const serviceRegex = getServiceRegex(
+    isLive() ? services : services.filter(service => service !== 'zhongwen'),
+  );
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/utils/index.js]: Revert 
```suggestion
  const serviceRegex = getServiceRegex(services);
```

File: src/app/routes/utils/regex/utils/index.js [hunk 5]
```diff
@@ -81,21 +81,25 @@ export const getFrontPageRegex = services => {
...
   return `/:service(${serviceRegex}):variant(${variantRegex})?/tipohome:amp(${ampRegex})?:lite(${liteRegex})?`;
 };
 
 export const getHomePageRegex = services => {
   let homePages = services;
   if (isLive()) {
     homePages = services.filter(service => homePageServices.includes(service));
+  } else {
+    homePages = services.filter(service => service !== 'zhongwen');
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/utils/index.js]: Revert
```suggestion
```

File: src/server/local/index.test.js [hunk 6]
```diff
@@ -20,7 +20,7 @@ describe('Local Server', () => {
     pageType                             | dataPath                                                         | expectedDataFile
     ${'Article'}                         | ${'/news/articles/c0g992jmmkko.json'}                            | ${'/news/articles/c0g992jmmkko.json'}
     ${'Article with variant'}            | ${'/zhongwen/articles/c3xd4x9prgyo/simp.json'}                   | ${'/zhongwen/articles/c3xd4x9prgyo/simp.json'}
-    ${'Migrated Home Page with variant'} | ${'/zhongwen/trad.json'}                                         | ${'/zhongwen/homePage/trad.json'}
+    ${'Migrated Home Page with variant'} | ${'/zhongwen/trad.json'}                                         | ${'/zhongwen/frontpage/trad.json'}
```
Comment (karinathomasbbc) [inline src/server/local/index.test.js]:
```suggestion
    ${'Migrated Home Page with variant'} | ${'/zhongwen/trad.json'}                                         | ${'/zhongwen/homePage/trad.json'}
```

File: src/app/lib/utilities/getStoryPromoInfo/index.jsx [hunk 7]
```diff
@@ -28,10 +28,15 @@ export const getHeadline = item => {
   return headline;
 };
 
-export const getUrl = item => {
+export const getUrl = (item, service = null, variant = null) => {
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getStoryPromoInfo/index.jsx]: Can we add a test for this to prove it will work OK?

**Review by karinathomasbbc — COMMENTED** [review]
Once the suggestions are reverted (because we don't want to go back to CPS frontpages on test), you should be able to test that the front page URLs have the same variant using:

`yarn build:live:debug && yarn start`

Then viewing the serbian/zhongwen homepage
http://localhost:7080/zhongwen/simp
 http://localhost:7080/zhongwen/trad

File: src/app/lib/utilities/getStoryPromoInfo/index.jsx [hunk 8]
```diff
@@ -28,10 +28,15 @@ export const getHeadline = item => {
...
   const assetUri = pathOr(null, ['locators', 'assetUri'], item);
   const canonicalUrl = pathOr(null, ['locators', 'canonicalUrl'], item);
-  const uri = pathOr(null, ['uri'], item);
+  let uri = pathOr(null, ['uri'], item);
+  if (uri && service === 'zhongwen') {
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getStoryPromoInfo/index.jsx]: Let's not restrict this to zhongwen only - why don't we check for the existence of variant instead?

File: src/app/lib/utilities/getStoryPromoInfo/index.jsx (hunk 1 above)
Comment (andrewscfc) [reply src/app/lib/utilities/getStoryPromoInfo/index.jsx]: I think the substring is agnostic of which service this is being applied on i.e. trad -> simp or cyr -> lat

A replace would need a specific string

**Review by amoore108 — APPROVED** [review]

**Review by karinathomasbbc — APPROVED** [review]
LGTM!

**Pull request merged**